core/trino-main/src/test/java/io/trino/sql/planner/BenchmarkPlanner.java

```diff
@@ -108,48 +98,84 @@ public void tearDown()
-            return benchmarkData.queries.stream()
-                    .map(query -> benchmarkData.queryRunner.createPlan(transactionSession, query, stage, false, WarningCollector.NOOP))
+            return benchmarkData.queries.getQueries().stream()
+                    .map(query -> benchmarkData.queryRunner.createPlan(transactionSession, query, benchmarkData.stage, false, WarningCollector.NOOP))
                     .collect(toImmutableList());
         });
     }
 
-    @Benchmark
-    public Plan planLargeInQuery(BenchmarkData benchmarkData)
-    {
-        return benchmarkData.queryRunner.inTransaction(transactionSession -> {
-            LogicalPlanner.Stage stage = LogicalPlanner.Stage.valueOf(benchmarkData.stage.toUpperCase(ENGLISH));
-            return benchmarkData.queryRunner.createPlan(
-                    transactionSession, benchmarkData.largeInQuery, stage, false, WarningCollector.NOOP);
-        });
-    }
-
     @Test
     public void verify()
     {
-        BenchmarkData data = new BenchmarkData();
-        data.setup();
         BenchmarkPlanner benchmark = new BenchmarkPlanner();
-        assertEquals(benchmark.planQueries(data).size(), 21);
-        assertNotNull(benchmark.planLargeInQuery(data));
+        for (Queries queries : Queries.values()) {
+            BenchmarkData data = new BenchmarkData();
+            data.queries = queries;
+            data.setup();
+            assertNotNull(benchmark.plan(data));
+        }
+    }
+
+    public static enum Queries
+    {
+        TPCH(() -> IntStream.rangeClosed(1, 22)
+                .boxed()
+                .filter(i -> i != 15) // q15 has two queries in it
+                .map(i -> readResource(format("/io/trino/tpch/queries/q%d.sql", i)))
+                .collect(toImmutableList())),
+        LARGE_IN("SELECT * from orders where o_orderkey in " +
+                IntStream.range(0, 5000)
+                        .mapToObj(Integer::toString)
+                        .collect(joining(", ", "(", ")"))),
+        // 86k columns present in the query with 500 group bys
+        GROUP_BY_WITH_MANY_REFERENCED_COLUMNS("WITH " + IntStream.rangeClosed(0, 500)
+                .mapToObj(i -> """
+                        t%s AS (
+                        SELECT * FROM lineitem a
+                        JOIN tiny.lineitem b ON a.l_orderkey = b.l_orderkey
+                        JOIN sf10.lineitem c ON a.l_orderkey = c.l_orderkey
+                        JOIN sf100.lineitem d ON a.l_orderkey = d.l_orderkey
+                        JOIN sf1000.lineitem e ON a.l_orderkey = e.l_orderkey
+                        WHERE a.l_orderkey = (SELECT max(o_orderkey) FROM orders GROUP BY o_orderkey))
+                        """
+                        .formatted(i))
+                .collect(joining(",")) +
+                "SELECT 1 FROM lineitem"),
+        /**/;
+
+        private final Supplier<List<String>> queries;
+
+        Queries(String query)
```

Comment:
let's ,make the others lazy too

Reply:
Not so easy. Constructors (Supplier<List<String>>) and (Supplier<String>) are identically typed because of the type erasure.